Sources/NativeBuilder/ContainerBuildSnapshotter/TarArchiveSnapshotter.swift

```diff
@@ -0,0 +1,360 @@
+        }
+        guard let currentLayerDigest = node.state.layerDigest else {
+            throw TarArchiveSnapshotterError.invalidSnapshotState("committed parent missing layer digest")
+        }
+
+        // If we've already materialized this layer during this run, return cached directory
+        if let cached = materializedBases[currentLayerDigest] {
+            return cached
+        }
+
+        // Destination path derived from the committed snapshot's layer digest
+        let baseRoot = cfg.workingRoot.appendingPathComponent("materialized", isDirectory: true)
+        let folderName = currentLayerDigest.replacingOccurrences(of: ":", with: "_")
+        let dest = baseRoot.appendingPathComponent(folderName, isDirectory: true)
+
+        // Reuse if already materialized on disk
+        var isDir: ObjCBool = false
+        if FileManager.default.fileExists(atPath: dest.path, isDirectory: &isDir), isDir.boolValue {
+            materializedBases[currentLayerDigest] = dest
+            return dest
+        }
+
+        // Ensure parent directory exists
+        try FileManager.default.createDirectory(at: baseRoot, withIntermediateDirectories: true)
+
+        // If there's a parent, ensure it is materialized first, then clone/copy it as our starting point.
+        if let p = node.parent {
+            guard case .committed = p.state else {
+                throw TarArchiveSnapshotterError.invalidSnapshotState("ancestor not committed")
+            }
+            guard p.state.layerDigest != nil else {
+                throw TarArchiveSnapshotterError.invalidSnapshotState("ancestor missing layer digest")
+            }
+
+            let parentDest = try await materializeCommitted(p)
+
+            // Ensure we start from a clean destination, then copy the parent's tree.
+            if FileManager.default.fileExists(atPath: dest.path) {
+                try FileManager.default.removeItem(at: dest)
+            }
+            try FileManager.default.copyItem(at: parentDest, to: dest)
+        } else {
+            // No parent: start from empty root directory
+            try FileManager.default.createDirectory(at: dest, withIntermediateDirectories: true)
+        }
+
+        // Fetch only THIS snapshot's layer blob and apply on top of prepared base
+        let store = cfg.differ.contentStore
+        guard let content = try await store.get(digest: currentLayerDigest) else {
+            throw TarArchiveSnapshotterError.invalidSnapshotState("missing content for layer \(currentLayerDigest)")
+        }
+        let bytes = try content.data()
+
+        // Heuristic: use declared media type if present, else default to gzip (our differ default)
+        let mediaType = node.state.layerMediaType ?? "application/vnd.oci.image.layer.v1.tar+gzip"
+
+        // Persist to a temp file for ArchiveReader
+        let suffix = mediaType.contains("+gzip") ? ".tar.gz" : ".tar"
```

Comment:
Could we reference DiffFormat's media type instead?

Reply:
This is the path of a temporary file being staged for archive reader